component/build_info.go

```diff
@@ -25,6 +25,11 @@ type BuildInfo struct {
 
 	// Version string.
 	Version string
+
+	// ServiceName is the value of "service.name" attribute in the telemetry
+	// that the Collector produces about itself. This can be overridden
+	// by the end user via telemetry.Config.Resource setting.
+	ServiceName string
```

Comment:
Why exposing it here? Why is not a config?

Reply:
It is also a config. I described in the PR description:

>I made the service.name a setting in BuildInfo which can be overridden when building the Collector. Other distros may choose a different service.name.
>The service.name can be also overridden by the end user via telemetry config setting.

Reply:
Why do you want to have it in the BuildInfo then?

Reply:
The BuildInfo defines the default value of `service.name` if it is unspecified by the end user in the config file. This allows distributions to choose their own default value of `service.name`. Do you think it is unnecessary? Alternatively, we can hard-code the service.name to `io.opentelemtry.collector`. Is it what you are suggesting?

Reply:
The BuildInfo is available to all components, is the intent to expose this to every component? If just to allow different distributions to configure we can do that in the `service.CollectorSettings`, but I don't think that is important. We should probably hardcode to `io.opentelemtry.collector` and see if any distribution wants to overwrite that and configuration is not enough.

Reply:
Sounds good. Let me change to hard-coded.

Reply:
Done.